lib/luadata.c

```diff
@@ -50,41 +51,76 @@ static inline luadata_t *luadata_checkdata(lua_State *L, lua_Integer *offset, lu
 	return data;
 }
 
-static int luadata_getnumber(lua_State *L)
-{
-	lua_Integer offset;
-	luadata_t *data = luadata_checkdata(L, &offset, LUADATA_NUMBER_SZ);
+#define LUADATA_NEWGETTER(what) 	\
+static int luadata_get##what(lua_State *L) \
+{					\
+	lua_Integer offset;		\
+	luadata_t *data = luadata_checkdata(L, &offset, sizeof(what##_t));	\
+					\
+	lua_pushinteger(L, (lua_Integer)*(what##_t *)(data->ptr + offset));	\
+	return 1;			\
+}
 
-	lua_pushinteger(L, *(lua_Integer *)(data->ptr + offset));
-	return 1;
+LUADATA_NEWGETTER(int8);
+LUADATA_NEWGETTER(uint8);
+LUADATA_NEWGETTER(int16);
+LUADATA_NEWGETTER(uint16);
+LUADATA_NEWGETTER(int32);
+LUADATA_NEWGETTER(uint32);
```

Comment:
let's move this to below `LUADATA_NEWSETTER` definition and declare getters and setters alternately.. the ifdef 64 will be cleaner in this way I guess.. e.g.,
```
LUADATA_NEWGETTER(int8);
LUADATA_NEWSETTER(int8);
...
```

actually, we can also have a macro that bundle both.. it will clean up the code a bit more I think.. e.g.,

```
#define LUADATA_NEWINT(T)	(LUADATA_NEWGETTER(T); LUADATA_NEWSETTER(T))
```